### PR TITLE
OCLOMRS-191: A user should be able to Release Dictionary HEAD as new version

### PR DESCRIPTION
**Pull request by kmunene**
# JIRA TICKET NAME:
[OCLOMRS-191: A user should be able to Release Dictionary HEAD as a new version
](https://issues.openmrs.org/browse/OCLOMRS-191)

# Summary:
Currently, a user is not able to release the HEAD (first) version of a dictionary. This issue should ensure that the owner of a dictionary can release its HEAD version.

### CODE REVIEW

File: src/tests/Dictionary/DictionaryContainer.test.jsx [hunk 1]
```diff
@@ -48,6 +48,67 @@ describe('DictionaryOverview', () => {
...
+      },
+      fetchDictionary: jest.fn(),
+      fetchVersions: jest.fn(),
+      fetchDictionaryConcepts: jest.fn(),
+      releaseHead: jest.fn(),
+    };
+    const wrapper = mount(<Provider store={store}><MemoryRouter><DictionaryOverview {...props} /></MemoryRouter></Provider>);
+    expect(wrapper.find('.fas.fa-cloud-upload-alt')).toHaveLength(1);
+    expect(wrapper).toMatchSnapshot();
+    wrapper.find('.fas.fa-cloud-upload-alt').simulate('click');
```
Comment (emmabaye) [inline src/tests/Dictionary/DictionaryContainer.test.jsx]: Great job. However, I think this line that simulates a click should come before the two assertions above it.

**Comment by dkayiwa** [comment]
@kmunene why has this taken an entire week without being reviewed and closed?

**Comment by kmunene** [comment]
@dkayiwa After putting it up for review there was feedback to change how I had implemented the task. I have however put it up for review.

File: src/tests/Dictionary/DictionaryContainer.test.jsx (hunk 1 above)
Comment (dondrzzy) [reply src/tests/Dictionary/DictionaryContainer.test.jsx]: Please review this test to simulate and test what happens when a user clicks what you are clicking.

File: src/tests/Dictionary/DictionaryContainer.test.jsx [hunk 2]
```diff
@@ -48,6 +48,67 @@ describe('DictionaryOverview', () => {
...
+          ownerType: 'testing',
+          owner: 'tester',
+          type: 'collection',
+          name: 'chris',
+        },
+      },
+      isReleased: true,
+      fetchVersions: jest.fn(),
+    };
+    wrapper.setProps(newProps);
```
Comment (dondrzzy) [inline src/tests/Dictionary/DictionaryContainer.test.jsx]: This test is not asserting anything. Please refactor accordingly.

File: src/tests/dictionaryConcepts/components/ConceptNameRows.test.jsx [hunk 3]
```diff
@@ -5,7 +5,7 @@ import ConceptNameRows from '../../../components/dictionaryConcepts/components/C
 const props = {
   pathName: {
     language: 'en',
-  }
```
Comment (dondrzzy) [inline src/tests/dictionaryConcepts/components/ConceptNameRows.test.jsx]: Is this a formatting change?

File: src/redux/actions/dictionaries/dictionaryActionCreators.js [hunk 4]
```diff
@@ -111,6 +112,24 @@ export const fetchDictionaryConcepts = data => (dispatch) => {
...
+        dispatch(realisingHeadSuccess(payload));
+        dispatch(isFetching(false));
+        notify.show(
+          'Head Version Successfully Released',
+          'success', 6000,
+        );
+      }
+    )
+    .catch(() => {
+          notify.show("Network Error", 'error', 6000);
```
Comment (dondrzzy) [inline src/redux/actions/dictionaries/dictionaryActionCreators.js]: Assuming `dispatch(isFetching(false))` stops the loader or the likes. Shouldn't you be doing the same here.

File: src/tests/dictionaryConcepts/components/ConceptNameRows.test.jsx (hunk 3 above)
Comment (kmunene) [reply src/tests/dictionaryConcepts/components/ConceptNameRows.test.jsx]: Yes it is.

File: src/tests/Dictionary/DictionaryContainer.test.jsx (hunk 1 above)
Comment (kmunene) [reply src/tests/Dictionary/DictionaryContainer.test.jsx]: Thanks for noticing this. Will work on it.

File: src/tests/Dictionary/DictionaryContainer.test.jsx (hunk 2 above)
Comment (kmunene) [reply src/tests/Dictionary/DictionaryContainer.test.jsx]: Thanks for noticing this. Will work on it.

File: src/redux/actions/dictionaries/dictionaryActionCreators.js (hunk 4 above)
Comment (kmunene) [reply src/redux/actions/dictionaries/dictionaryActionCreators.js]: Thanks for the observation. I had missed that.

**Comment by kmunene** [comment]
@alexmochu Thank you for the observation. I have made the change.

File: src/tests/dictionaryConcepts/components/ConceptNameRows.test.jsx (hunk 3 above)
Comment (dondrzzy) [reply src/tests/dictionaryConcepts/components/ConceptNameRows.test.jsx]: All formatting changes that are not in line with your PR task should be done in a different PR.

Comment (kmunene) [reply src/tests/dictionaryConcepts/components/ConceptNameRows.test.jsx]: That is true!

File: src/redux/actions/dictionaries/dictionaryActionCreators.js [hunk 5]
```diff
@@ -111,6 +112,25 @@ export const fetchDictionaryConcepts = data => (dispatch) => {
...
+        dispatch(isFetching(false));
+        notify.show(
+          'Head Version Successfully Released',
+          'success', 6000,
+        );
+      }
+    )
+    .catch(() => {
+      dispatch(isFetching(false));
+      notify.show("Network Error. Please try again later!", 'error', 6000);
```
Comment (dondrzzy) [inline src/redux/actions/dictionaries/dictionaryActionCreators.js]: @kmunene, this seems to be a put request which seems like one that requires the authorization of some sort. Is this the only error possibility, `network error`?

Comment (kmunene) [reply src/redux/actions/dictionaries/dictionaryActionCreators.js]: If one is either not logged in or not the owner of that dictionary they are not able to view the "Release Head Version" so they cannot make the put request.

**Comment by dkayiwa** [comment]
Can you look into these merge conflicts?

**Pull request merged**